docs/coroutine-context-and-dispatchers.md

```diff
@@ -487,48 +487,43 @@ I'm working in thread DefaultDispatcher-worker-1 @test#2
 
 <!--- TEST FLEXIBLE_THREAD -->
 
-### Cancellation via explicit job
+### Getting it together
 
 Let us put our knowledge about contexts, children and jobs together. Assume that our application has
 an object with a lifecycle, but that object is not a coroutine. For example, we are writing an Android application
 and launch various coroutines in the context of an Android activity to perform asynchronous operations to fetch 
 and update data, do animations, etc. All of these coroutines must be cancelled when activity is destroyed
-to avoid memory leaks. 
-  
-We manage a lifecycle of our coroutines by creating an instance of [Job] that is tied to 
-the lifecycle of our activity. A job instance is created using [Job()] factory function when
-activity is created and it is cancelled when an activity is destroyed like this:
+to avoid memory leaks. We, of course, can manipulate contexts and jobs manually to tie activity's 
+and coroutines lifecycles, but `kotlinx.coroutines` provides an abstraction that encapsulates that: [CoroutineScope].
+You should be already familiar with coroutine scope as all coroutine builders are declared as extensions on it. 
 
+We manage a lifecycle of our coroutines by creating an instance of [CoroutineScope] that is tied to 
+the lifecycle of our activity. `CoroutineScope` instance can be created by [CoroutineScope()] or [MainScope()]
+factory functions. The former creates a general-purpose scope, while the latter creates scope for UI applications and uses
+[Dispatchers.Main] as default dispatcher:
 
 <div class="sample" markdown="1" theme="idea" data-highlight-only>
 
 ```kotlin
-class Activity : CoroutineScope {
-    lateinit var job: Job
-
-    fun create() {
-        job = Job()
-    }
-
+class Activity {
+    private val mainScope = MainScope()
+    
     fun destroy() {
-        job.cancel()
+        mainScope.cancel()
     }
     // to be continued ...
 ```
 
 </div>
 
-We also implement [CoroutineScope] interface in this `Actvity` class. We only need to provide an override
-for its [CoroutineScope.coroutineContext] property to specify the context for coroutines launched in its
-scope. We combine the desired dispatcher (we used [Dispatchers.Default] in this example) and a job:
+We can implement [CoroutineScope] interface in this `Actvity` class. The best way to do it is
+to use delegation with default factory functions.
+We also can combine the desired dispatcher (we used [Dispatchers.Default] in this example) with the scope:
 
 <div class="sample" markdown="1" theme="idea" data-highlight-only>
 
 ```kotlin
-    // class Activity continues
-    override val coroutineContext: CoroutineContext
-        get() = Dispatchers.Default + job
-    // to be continued ...
+    class Activity : CoroutineScope by CoroutineScope(Dispatchers.Default)
```

Comment:
I'd suggest to add opening curly brace and `// to be continued ...` like in the previous version of `Activity` class for consistency.